components/navigation/SideBar.tsx

```diff
@@ -45,6 +47,84 @@ const MenuLink: React.FC<MenuLinkProps> = ({
   );
 };
 
+type DashboardLinkProps = {
+  siteName: string;
+  linkProps: LinkProps;
+  currentPath: string;
+  siteUUID?: string;
+  allSiteUUID?: string[];
+};
+
+const DashboardLink: React.FC<DashboardLinkProps> = ({
+  linkProps,
+  siteName,
+  currentPath,
+  siteUUID,
+  allSiteUUID,
+}) => {
+  const textColor =
+    linkProps.href === currentPath ? 'text-white' : 'text-ocf-gray-800';
+  const borderColor =
+    linkProps.href === currentPath ? 'border-amber' : 'border-ocf-gray-1000';
+
+  const { forecastData } = useSiteData(siteUUID || '');
+  const { totalExpectedGeneration } = useSiteAggregation(allSiteUUID || []);
+
+  const currentOutput = useMemo(() => {
+    if (allSiteUUID) {
+      let d = totalExpectedGeneration
+        ? Math.round(getCurrentTimeGeneration(totalExpectedGeneration) * 100) /
+          100
+        : undefined;
+      return d;
+    }
+    return forecastData
+      ? Math.round(
+          getCurrentTimeGeneration(forecastData.forecast_values) * 100
+        ) / 100
+      : undefined;
+  }, [forecastData, allSiteUUID, totalExpectedGeneration]);
+
+  const generateThresholdGraph = () => {
+    let suitUUIDToUse = null;
+    if (allSiteUUID !== undefined && allSiteUUID.length) {
+      suitUUIDToUse = allSiteUUID[0];
+    } else if (siteUUID !== undefined) {
+      suitUUIDToUse = siteUUID;
+    }
+
+    if (suitUUIDToUse === null) {
+      return null;
+    }
+
+    return <SiteGraph siteUUID={suitUUIDToUse} hidden={false} height={50} />;
+  };
+
+  return (
+    <Link {...linkProps}>
```

Comment:
I think this `<Link` isn't rendering properly because Next.js is weird. Apply the props `passHref` and then make this have a first child of an `<a>` and wrap all children in it.